cc/raster/one_copy_raster_buffer_provider.cc

```diff
@@ -362,9 +362,13 @@ void OneCopyRasterBufferProvider::PlaybackToStagingBuffer(
 #if defined(CASTANETS)
     if (std::string("renderer") ==
         base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII("type")) {
+      mojo::CompressionMode compression_mode = mojo::CompressionMode::ZLIB;
+      if (staging_buffer->size.width() == 256 &&
+          staging_buffer->size.height() == 256)
+        compression_mode = mojo::CompressionMode::WEBP;
       mojo::SyncSharedMemory(
           buffer->GetHandle().handle.GetGUID(), 0,
-          staging_buffer->size.width() * staging_buffer->size.height() * 4);
+          staging_buffer->size.width() * staging_buffer->size.height() * 4, compression_mode);
     }
```

Comment:
Instead of passing compression_mode(uint32_t), how about passing a class or a structure that includes tile size and types?

Reply:
On the latest patch set, we pass the required data as a struct to use webp for all tiles.

Reply:
How about using SyncSharedMemory2d() instead of SyncSharedMemory() for bitmap.
I think we can also support full bitmap(offset is zero and stride is same as width) in SyncSharedMemory2d() .

Reply:
Ok we can use SyncSharedMemory2d. I have some clarification, 

1. I tried to test SyncSharedMemory2d for partial tiles, but it looks the module is not working properly, it looks there is stride/width is not proper. Only after I modify as below I could see proper unbroken tiles, else data corruption happens. Can you please clarify whether my understanding is wrong.
https://github.com/suyambulingamrm/Castanets/commit/c881318e3914c05be96ab741f54689502f46501a

2. full bitmap(offset is zero and stride is same as width) => stride (rowbytes) should be width * bytesperpixel ?

Reply:
https://github.com/Samsung/Castanets/pull/205/commits/c9f55f1b755f7eb4a4464867d5de464ad984f2f1
Yes, Sunwoo intended width, height, offset as bytes, not pixel.
Currently, this function seems to be not used anymore. How about changing function parameter for webp compression?

Reply:
Oh ok.. Changing the Sync2d API as mentioned below is ok ? we need height info of the rect as well for webp encode api.
(From) SyncSharedMemory2d(const base::UnguessableToken& guid,
                   size_t offset,
                   size_t sync_size,
                   size_t width,
                   size_t stride,
                   BrokerCompressionData compression_data=BrokerCompressionData());
(To) SyncSharedMemory2d(const base::UnguessableToken& guid,
                   size_t offset,
  		   size_t width,
  		   size_t height,
                   size_t bytesperpixel,
                   size_t stride,
		   CompressionMode mode);

Reply:
LGTM. But I prefer the following order : size_t width, size_t height, size_t bpp, size_t offset, size_t stride, CompressionMode mode
And I think pixel is better than byte for width, height, offset, and stride.

Reply:
Updated on latest patchset to use 2d api for tiles as suggested.